data/src/main/java/io/gnosis/data/models/Transaction.kt

```diff
@@ -37,7 +37,8 @@ sealed class Transaction {
         val value: BigInteger,
         val date: String?,
         val tokenInfo: ServiceTokenInfo?,
-        val nonce: BigInteger?
+        val nonce: BigInteger?,
+        val incoming: Boolean
```

Comment:
Not sure if it's a good idea to add this as part of data model, as the client gateway will return this hopefully with need for mapping. Instead of adding an additional field could we just have it calculated as an extension function? Like `fun Transactions.Transfer.isInconming(): Boolean`? This is a field that to me is part of the UI layer.

Reply:
After seeing the rest of the PR, the `incoming` calculation should definitely be moved up. The basic mapping should be as clean as possible, meaning, with fewer dependencies as possible (safeInfo and other stuff)

Reply:
We already had a dependency on safeInfo for the status calculation.

Reply:
Why is the status calculation here and not in the UI?

Reply:
The status calculation will come from the gateway too.

Reply:
Should the incoming calculation also be done in the gateway (in the future)?

Reply:
Since we just agreed, to have an incoming field coming from the client gateway at some point. Can this PR be merged and the code for incoming calculation be removed when the field is returned by the gateway?